pydm/widgets/scatterplot.py

```diff
@@ -80,13 +84,39 @@ def x_address(self, new_address):
         -------
         new_address: str
         """
-        if new_address is None or len(str(new_address)) < 1:
-            self.x_channel = None
+        if new_address is None or len(new_address) == 0:
```

Comment:
This entire method looks fairly idiomatic, and something that may be desirable to make a utility function out of (I could be wrong, though).

Something like: `address_setter_helper(self, 'x_channel', new_address)`